server/api/controllers/image-controller.ts

```diff
@@ -0,0 +1,271 @@
+import { Request, Response } from "express";
+import { UploadedFile } from "express-fileupload";
+import { Logger } from "../../util/logger";
+import { Image } from "../../types/image";
+import {
+  Dirent,
+  existsSync,
+  lstatSync,
+  readdirSync,
+  readFileSync,
+  unlinkSync,
+} from "fs";
+import { ReadWriteController } from "./read-write-controller";
+import path from "path";
+
+enum Action {
+  UPLOAD,
+  DELETE,
+}
+
+type ImageRequest = Image & {
+  res: Response;
+};
+
+type ImageUploadRequest = ImageRequest & {
+  type: Action.UPLOAD;
+  uploadedFile: UploadedFile | null;
+};
+
+type ImageDeleteRequest = ImageRequest & {
+  type: Action.DELETE;
+};
+
+enum ImageMimetypes {
+  JPEG = "image/jpeg",
+  PNG = "image/png",
+}
+
+export class ImageController {
+  static logger = new Logger();
+  private static queues: Record<string, ImageRequestQueue> = {};
+
+  static uploadImage(req: Request, res: Response) {
```

Comment:
These functions make a lot of sense now - would it be possible to add some brief comments outlining each one to help us debug one month in the future when we've forgotten what they do 😅